draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -717,6 +737,175 @@ Steps:
 23. Output (x, y)
 ~~~
 
+### Boneh-Franklin Method {#supersingular}
+
+The map2curve_bf(alpha) implements the Boneh-Franklin method {{BF01}} which
+covers the case of supersingular curves E: y^2=x^3+B.
+
+**Preconditions**
+
+This algorithm works for any Weierstrass curve over F_{q} such that A=0 and
+q=2 mod 3.
+
+**Examples**
+
+- y^2=x^3+1
+
+**Algorithm**: map2curve_bf
+
+Input:
+
+ - alpha: an octet string to be hashed.
+ - B    : the constant from the Weierstrass curve.
+
+Output:
+
+ - (x,y), a point in E.
+
+Operations:
+
+~~~
+1. u = HashToBase(alpha)
+2. x = (u^2-B)^(1/3)
+3. Output (x,u)
+~~~
+
+**Implementation**
+
+The following procedure implements the Boneh-Franklin's algorithm in a
+straight-line fashion.
+
+~~~
+map2curve_bf(alpha)
+
+Input:
+
+  alpha - an octet string to be encoded
+
+Output:
+
+  (x, y) - a point in E
+
+Precomputations:
+
+1.  c = (2p-1)/3           // Integer arithmetic
+
+Steps:
+
+1.  u = HashToBase(alpha)  // {0,1}^* -> Fp
+2. t0 = u^2                // t0 = u^2
+3. t1 = t0-B               // t1 = u^2-B
+4.  x = t1^c               // x  = (u^2-B)^(1/3)
+5. Output (x,u)
+~~~
+
+
+### Fouque-Tibouchi Method {#ftpairing}
+
+The map2curve_ft(alpha) implements the Fouque-Tibouchi method {{FT12}} which
+covers the case of pairing-friendly curves E: y^2=x^3+B.
+Note that for pairing curves the destination group is usually a subgroup of the
+curve, hence, a scalar multiplication by the cofactor will send the point to
+the desired group.
+
+**Preconditions**
+
+This algorithm works for any Weierstrass curve over F_{q} such that A=0, 1+B is
+a non-zero square in the field, and q=7 mod 12 (covering the case q=1 mod 3 not
+handled by Boneh-Franklin's method).
+
+**Examples**
+
+- SECP256K1 curve
+- BN curves
+- KSS curves
+- BLS curves
+
+**Algorithm**: map2curve_ft
+
+Input:
+
+ - alpha: an octet string to be hashed.
+ - B    : the constant from the Weierstrass curve.
+ - s    : a constant equal to sqrt(-3)
+
+Output:
+
+ - (x,y), a point in E.
+
+Operations:
+
+~~~
+1.  Define f(x) = x^3+B
```

Comment:
Why is this an operation? Can we lift this up to somewhere above?

Reply:
Removed and explicitly stated.